src/xdpinstaller/scripts/xdp-setup.ps1

```diff
@@ -0,0 +1,226 @@
+<#
+
+.SYNOPSIS
+This script installs or uninstalls various XDP components.
+
+.PARAMETER Install
+    Specifies an XDP component to install.
+
+.PARAMETER Uninstall
+    Attempts to uninstall all XDP components.
+
+#>
+
+param (
+    
+    [Parameter(Mandatory = $false)]
+    [ValidateSet("", "xdp")]
+    [string]$Install = "",
+
+    [Parameter(Mandatory = $false)]
+    [ValidateSet("", "xdp")]
+    [string]$Uninstall = ""
+)
+
+Set-StrictMode -Version 'Latest'
+$PSDefaultParameterValues['*:ErrorAction'] = 'Stop'
+
+# Global paths.
+$SystemFolder = [Environment]::SystemDirectory
+$XdpSys = "$SystemFolder\drivers\xdp.sys"
+$XdpInf = "$SystemFolder\drivers\xdp.inf"
+$XdpCat = "$SystemFolder\drivers\xdp.cat"
+$LogsDir = "$SystemFolder\Logs"
+
+# Set the temporary working directory and testore it on exit.
+Push-Location -Path $SystemFolder
+
+# Helper to reboot the machine.
+function Uninstall-Failure {
+    Write-Host "Capturing live kernel dump"
+
+    New-Item -ItemType Directory -Force -Path $LogsDir | Out-Null
+    Write-Verbose "$LiveKD -o $LogsDir\xdp.dmp -k $KD -ml -accepteula"
+    & $LiveKD -o $LogsDir\xdp.dmp -k $KD -ml -accepteula
+
+    Write-Host "##vso[task.setvariable variable=NeedsReboot]true"
+    Write-Error "Preparing to reboot machine!"
+}
+
+# Helper to start (with retry) a service.
+function Start-Service-With-Retry($Name) {
```

Comment:
We can remove this if we agree that relying on the netcfg install to start things is good enough.

Reply:
Agreed. It is better to fail once and detect unreliability at a higher level.

Reply:
You mean, attempt to start the sevice once, or not starting it at all (i.e. as per our internal sync)?

Reply:
I would rely on netcfg which should attempt to auto-start the driver